### ISSUE TITLE
Error: input UTxOs cannot cover the transaction outputs and tx fee

### ISSUE THREAD

**Issue by robinboening** [Python-Cardano/pycardano#26]
First of all, thank you for this awesome library! You guys rock!

I am trying to mint some NFT and the [example](https://github.com/cffls/pycardano/blob/main/examples/native_token.py) is failing for me. I have to believe its a bug and not a user error since I haven't altered any of the code.

**Describe the bug**
The [example for native token minting](https://github.com/cffls/pycardano/blob/main/examples/native_token.py) is failing with an error due to mismatching ADA amounts in the change output and the fee.

Input: 1000 ADA
Change: 998.355763 ADA
Fee: 2.520477 ADA

Fee + Change = 1000.87624 ADA

**To Reproduce**
Run the original example script, top up the generated payment address using the faucet and run again
https://github.com/cffls/pycardano/blob/main/examples/native_token.py

**Logs**
```
Traceback (most recent call last):
  File "/Users/robin/projects/../mint_example.py", line 162, in <module>
    signed_tx = builder.build_and_sign([payment_skey, policy_skey], change_address=address)
  File "/usr/local/lib/python3.9/site-packages/pycardano/txbuilder.py", line 691, in build_and_sign
    tx_body = self.build(change_address=change_address)
  File "/usr/local/lib/python3.9/site-packages/pycardano/txbuilder.py", line 667, in build
    self._add_change_and_fee(change_address)
  File "/usr/local/lib/python3.9/site-packages/pycardano/txbuilder.py", line 351, in _add_change_and_fee
    changes = self._calc_change(
  File "/usr/local/lib/python3.9/site-packages/pycardano/txbuilder.py", line 286, in _calc_change
    raise InvalidTransactionException(
pycardano.exception.InvalidTransactionException: The input UTxOs cannot cover the transaction outputs and tx fee.
Inputs: [{'input': {'index': 0,
 'transaction_id': TransactionId(hex='009c527e...9b2e1')},
 'output': {'address': addr_test1vz...,
 'amount': 1000000000,
 'datum_hash': None}}]
Outputs: [{'address': addr_test1vz...,
 'amount': {'coin': 1448244,
 'multi_asset': {ScriptHash(hex='90ddf27...981c8'): {AssetName(b'MY_NFT_1'): 1, AssetName(b'MY_NFT_2'): 1}}},
 'datum_hash': None}, {'address': addr_test1vz...,
 'amount': 998355763,
 'datum_hash': None}]
fee: 2520477
```

**Expected behavior**
The Change/Fee calculation should match the Input and the Tx should be valid

**Environment and software version (please complete the following information):**
 - OS: Mac OS 10.15
 - PyCardano Version 0.3.1

Thanks!

**Comment by cffls** [comment]
Hi Robin,

Thank you for reporting this issue. It is indeed a bug from the example. After some testing, I found that the example built the transaction twice (a mistake from a previous refactoring 😅 ) . I removed [this unnecessary line](https://github.com/cffls/pycardano/blob/14d6a38582862d187830c35846eaff0c0cb493cf/examples/native_token.py#L162) from the example. It should work again now. Sorry about the confusion and thanks again for reporting this.

Jerry
